data-prepper-pipeline-parser/src/main/java/org/opensearch/dataprepper/pipeline/parser/transformer/TransformersFactory.java

```diff
@@ -41,35 +53,63 @@ public String getTransformationRulesDirectoryLocation() {
     }
 
     public String getPluginTemplateFileLocation(String pluginName) {
-        if(pluginName == null || pluginName.isEmpty()){
-            throw  new RuntimeException("Transformation plugin not found");
+        if (pluginName == null || pluginName.isEmpty()) {
+            throw new RuntimeException("Transformation plugin not found");
         }
         return templatesDirectoryPath + "/" + pluginName + TEMPLATE_FILE_NAME_PATTERN;
     }
 
-    public String getPluginRuleFileLocation(String pluginName) {
-        if(pluginName == null || pluginName.isEmpty()){
-            throw  new RuntimeException("Transformation plugin not found");
+    public InputStream getPluginTemplateFileStream(String pluginName) {
+        if (pluginName == null || pluginName.isEmpty()) {
+            throw new RuntimeException("Transformation plugin not found");
         }
-        return rulesDirectoryPath + "/" + pluginName + RULE_FILE_NAME_PATTERN;
+        ClassLoader classLoader = TransformersFactory.class.getClassLoader();
+        InputStream filestream = classLoader.getResourceAsStream("templates" + "/" + pluginName + TEMPLATE_FILE_NAME_PATTERN);
+        return filestream;
     }
 
-    public InputStream getPluginRuleFileStream(String pluginName) {
-        if(pluginName == null || pluginName.isEmpty()){
-            throw  new RuntimeException("Transformation plugin not found");
+    public List<Path> getRuleFiles() {
+        // Get the URI of the rules folder
+        URI uri = null;
+        try {
+            uri = getClass().getClassLoader().getResource("rules").toURI();
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+
+        Path rulesFolderPath;
+
+        if ("jar".equals(uri.getScheme())) {
+            // File is inside a JAR, create a filesystem for it
+            try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.emptyMap())) {
+                rulesFolderPath = fileSystem.getPath("rules");
+                return scanFolder(rulesFolderPath);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        } else {
+            // File is not inside a JAR
+            rulesFolderPath = Paths.get(uri);
```

Comment:
When would this ever occur?

Reply:
I was trying to address 2 cases:
1. "jar" scheme: This case occurs when the code is running from within a JAR file, ie during runtime.
2. Non-"jar" scheme: This case occurs when the code is running in a development environment where the rules directory is available in the file system and not within a jar.